src/raven/client.clj

```diff
@@ -385,6 +387,38 @@
     :data data
     :env env}))
 
+(defn get-full-ring-url
+  "Given a ring compliant request object, return the full URL.
+    This was lifted from ring's source code so as to not depend on it."
+  [request]
+  (str (-> request :scheme name)
+       "://"
+       (get-in request [:headers "host"])
+       (:uri request)
+       (when-let [query (:query-string request)]
+         (str "?" query))))
+
+(defn get-ring-env
+  [request]
+  (cond-> {:REMOTE_ADDR (:remote-addr request)
+           :websocket? (:websocket? request)
+           :route-params (:route-params request)
+           :params (:params request)}
+    (:compojure/route request) (assoc :compojure/route (:compojure/route request))))
```

Comment:
Use `some?` when testing for presence, *e.g*: `(some? (:compojure/route request) (assoc ...))`
Also, for bidi users, another pair would help:

```clojure
(some? (:route request) (assoc :route (get-in request [:route :handler])))
```

Reply:
Nice!

Thanks for the TIL about `some?`. I also added the bidi info as :bidi/route